src/Nethermind/Nethermind.Mev/Data/SimulatedMevBundle.cs

```diff
@@ -15,34 +15,74 @@
 //  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
 // 
 
+using System;
+using System.Linq;
 using Nethermind.Int256;
+using Nethermind.TxPool;
+using Nethermind.Core;
+using System.Collections.Generic;
+using Nethermind.Core.Crypto;
 
 namespace Nethermind.Mev.Data
 {
     public class SimulatedMevBundle
     {
-        public SimulatedMevBundle(MevBundle bundle, bool success, long gasUsed, UInt256 txFees, UInt256 coinbasePayments)
+
+        public SimulatedMevBundle(MevBundle bundle, 
+            bool success,
+            ICollection<Keccak> failedTransactions, 
+            long gasUsed, 
+            UInt256 txFees, 
+            UInt256 coinbasePayments, 
+            Dictionary<Keccak, UInt256> eligibleGasFeePaymentPerTransaction)
         {
             Bundle = bundle;
             Success = success;
+            FailedTransactions = failedTransactions;
             GasUsed = gasUsed;
             TxFees = txFees;
             CoinbasePayments = coinbasePayments;
+            EligibleGasFeePaymentPerTransaction = eligibleGasFeePaymentPerTransaction;
+            IEnumerable<Keccak?> unexpectedFailedTransactions = failedTransactions.Where(tx => !bundle.RevertingTxHashes.Contains(tx));
+            if (!unexpectedFailedTransactions.Any() && Success == false)
+            {
+                Success = true;
+            }
         }
 
         public UInt256 CoinbasePayments { get; set; }
         
         public UInt256 TxFees { get; set; }
-
+        
+        public Dictionary<Keccak, UInt256> EligibleGasFeePaymentPerTransaction { get; set; }
+        
+        public UInt256 EligibleGasFeePayment
+        {
+            get
+            {
+                // UInt256 doesn't implement Sum for some reason? need to clean this up
+                UInt256 sum = UInt256.Zero;
+                foreach (UInt256 value in EligibleGasFeePaymentPerTransaction.Values)
+                {
+                    sum += value;
+                }
+                return sum;
+            }
+        }
         public UInt256 Profit => TxFees + CoinbasePayments;
 
         public MevBundle Bundle { get; }
+        
         public bool Success { get; }
 
+        public ICollection<Keccak> FailedTransactions { get; }
```

Comment:
Change it to byte[] or BitVector, holding true/false for each transaction and map to Transactions jus by index